### PR TITLE
[RADOS] Core-workflow modules to control and modify QoS mClock parameters

### PR DESCRIPTION
**Pull request by harshkumarRH**
The default OSD QoS was changed to mClock with the RHCS 6.1 release, with this change, most of the recovery, backfilling and other background operations were now controlled by mClock and could be tuned by the customer as per their needs.

Scope of this PR is to add modules in core-workflows to facilitate the modification of mClock's parameter's default values.

Existing modules modified:
- _set_mclock_profile_
- _list_orch_services_

New modules added:
- _check_osd_op_queue_
- _set_mclock_parameter_
- _restart_daemon_services_

Pacific
Test suite: suites/pacific/rados/tier-2_rados_test-osd-rebalance.yaml
Pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EUYKT8/

Quincy
Test suite: suites/quincy/rados/tier-2_rados_test-osd-rebalance.yaml
Pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-M7WLTL/

Reef
Test suite: suites/reef/rados/tier-2_rados_test-osd-rebalance.yaml
Pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-TB3EUN

Signed-off-by: Harsh Kumar <hakumar@redhat.com>

### CODE REVIEW

File: ceph/rados/core_workflows.py [hunk 1]
```diff
@@ -1301,8 +1301,60 @@ def set_mclock_profile(self, profile, osd="osd"):
         Args:
             profile: mclock profile name
             osd: "osd" service by default or "osd.<Id>"
+            reset: revert mClock profile to default - balanced
         """
-        return self.node.shell([f"ceph config set {osd} osd_mclock_profile {profile}"])
+        if self.check_osd_op_queue(qos="mclock"):
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:1306]: Can we add a check here to see if we are in 6.1 and above, while we test this?

File: ceph/rados/core_workflows.py [hunk 2]
```diff
@@ -1301,8 +1301,60 @@ def set_mclock_profile(self, profile, osd="osd"):
...
+
+    def set_mclock_parameter(self, param: str, value, restart_osd: bool = False):
+        """Set value for any of the valid mClock config parameters
+        Args:
+            param (str): mClock config parameter to be modified
+            value: value to be set for the input parameter
+            restart_osd (boolean): flag to control restart of all OSDs;
+                necessary only for few parameters, hence added as a tunable setting.
+        """
+        if self.check_osd_op_queue(qos="mclock"):
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:1341]: same as above.

File: ceph/rados/core_workflows.py [hunk 3]
```diff
@@ -1772,3 +1824,67 @@ def get_ceph_pg_dump(self, pg_id: str) -> dict:
 
         log.error(f"PG {pg_id} not found in ceph pg dump output")
         raise KeyError(f"PG {pg_id} not found in ceph pg dump output")
+
+    def restart_daemon_services(self, daemon: str):
```
Comment (pdhiran) [inline ceph/rados/core_workflows.py:1828]: should we have this module to start, stop and restart? Looks good for now,  We can have enhancements in the next PR.

File: ceph/rados/core_workflows.py (hunk 1 above)
Comment (harshkumarRH) [reply ceph/rados/core_workflows.py]: Added a check to pass the execution if RHCS build is below 6

File: ceph/rados/core_workflows.py (hunk 2 above)
Comment (harshkumarRH) [reply ceph/rados/core_workflows.py]: Covered

File: ceph/rados/core_workflows.py (hunk 3 above)
Comment (harshkumarRH) [reply ceph/rados/core_workflows.py:1843]: Noted, will create a backlog task

**Review by pdhiran — APPROVED** [review]
Looks good

**Pull request merged**